app/views/courses/TeacherClassesView.coffee

```diff
@@ -71,6 +129,45 @@ module.exports = class TeacherClassesView extends RootView
 
   onLoaded: ->
     helper.calculateDots(@classrooms, @courses, @courseInstances)
+
+    @teacherQuestData['create_classroom'].complete = @classrooms.length > 0
+    for classroom in @classrooms.models
+      continue unless classroom.get('members')?.length > 0
+      classCompletion = {}
+      classCompletion[key] = 0 for key in Object.keys(@teacherQuestData)
+      students = classroom.get('members')?.length
+
+      
+      kithgardGatesCompletes = 0
+      wakkaMaulCompletes = 0
+      for session in classroom.sessions.models
+        if session.get('level')?.original is '541c9a30c6362edfb0f34479' # kithgard-gates
+          ++classCompletion['kithgard_gates_100']
+        if session.get('level')?.original is '5630eab0c0fcbd86057cc2f8' # wakka-maul
+          ++classCompletion['wakka_maul_100']            
+        continue unless session.get('state')?.complete
```

Comment:
In general I prefer more explicit control structures than procedural equivalents that depend strongly on instruction order, since explicit control structures are easier to interpret (especially when code starts getting more complicated)
```
if session.get('state')?.complete
  ...foo
```